docs/src/ref_impl/installer.md

```diff
@@ -5,67 +5,66 @@
-          echo "Now you can detach installation device and reboot to Ghaf."
-      else
-          echo "Some error occured during flushing process, exit code: $?."
-          exit
-      fi
-    '';
-  };
-}
-```
+Ghaf does not have conventional (imperative, non-declarative) installer, which
+you might expect in most of the conventional linux distros. Instead of this we
+have nixos-like non-interactive, declarative. In other words instead of clicking
+similar options each time you try to install system, you'll need to configure
+system once and deploy this configuraiton to desired machines. This done in a
+way that you need to use [ghaf as a library](./ghaf-based-project.md) in order to install system.
+
+This done via awesome tool
+[nixos-anywhere](https://github.com/nix-community/nixos-anywhere). There are two
+separate ways to use it: manually (by following 
+[their guide](https://nix-community.github.io/nixos-anywhere/quickstart.html)),
+or by using ghaf installation wizard.
+
+If you aware of NixOS installation process then this approach
+will seem familiar. Basically they are very similar, except system configuration
+is done in the familiar environment.
+
+## Via wizard
+
+Ghaf wizard will guide you through the process of creation of system
+configuration that you need.
+
+## Manual
+
+For now manual way is the only way to install ghaf, as wizard currently under
+development.
+
+To install ghaf manuall you need to:
 
-After that you can build an installer image using this command:
+1. Create your own flake via ghaf template
 
 ```sh
-nix build .#nixosConfigurations.<CONFIGURATION>.config.system.build.installer
+nix flake init -t github:tiiuae/ghaf#target-x86_64-generic
 ```
 
-## Adding Installer Modules
+2. Edit it according to your preferences.
 
-To add an installer module, replace the corresponding placeholders with your code and add this to your configuraiton:
+3. Set value of `ghaf.installer.sshKeys` in order to get installer image.
 
-```nix
-ghaf.installer.installerModules.<MODULE_NAME> = {
-  requestCode = ''
-    # Your request code written in Bash
-  '';
-  providedVariables = {
-    # Notice the dollar sign before the actual variable name in Bash.
-    <VARIABLE_NAME> = "$<VARIABLE_NAME>";
-  };
-};
+4. Build installer image.
+
+```sh
+nix build .#nixosConfigurations.PROJ_NAME-ghaf-debug.config.system.build.installer
 ```
 
-## Built-in Installer Modules
+5. Flush it to your device.
 
-Provided variables show variable names in Nix. For actual names of variables in Bash, see the sources of the module.
+```sh
+sudo dd if=./result/iso/nixos-...-linux.iso of=/dev/YOUR_DEVICE conv=sync && sync
+```
 
-### flushImage
+6. Run it on the hardware.
 
-Provided variables:
+7. Connect to the network (via `wifi-connector`).
 
-- deviceName: name of the device on which image should be flushed (e.g. "sda", "nvme0n1")
+8. Get block device name and put it in disko configuration (can be done via `lsblk` command).
```

Comment:
Get a block device name and put it in a disk configuration using the `lsblk` command.